libnavui-androidauto/build.gradle

```diff
@@ -45,7 +45,8 @@ dependencies {
     def carSearchVersion = "1.0.0-beta.35"
     api("com.mapbox.navigation:android:${carNavVersion}")
     api("com.mapbox.search:mapbox-search-android:${carSearchVersion}")
-    implementation("com.mapbox.navigation:ui-app:${carNavVersion}")
+    //implementation("com.mapbox.navigation:ui-app:${carNavVersion}")
+    implementation project(":libnavui-app")
```

Comment:
Revert this and upgrade dependency to `v2.8.0-beta.3` when released.

Reply:
Why did this need to be done though? Did this create an incompatibility.

The goal is to make it so different versions of the nav-sdk can be used with different versions of android-auto. When there is a hard dependency, we should identify it and build a stable abstraction

Reply:
It is because an incompatible change with `SharedApp` (which is in the internal package) was introduced in this PR. If we want to avoid such issues in the future, we should use only public APIs or maintain compatibility of all internal APIs that Android Auto uses.

Reply:
`SharedApp` needs to have a backwards compatible API, I don't see why it would need to be public.

I've reverted to see what the error is. This is because the sdk does not have a version so maven is unable to upgrade it. I think we could fix it if the sdk was built as a next version snapshot.
```
    java.lang.NoSuchMethodError: No static method setup$default(Lcom/mapbox/navigation/ui/app/internal/SharedApp;Landroid/content/Context;Lcom/mapbox/navigation/ui/voice/internal/MapboxAudioGuidance;ILjava/lang/Object;)V in class Lcom/mapbox/navigation/ui/app/internal/SharedApp; or its super classes (declaration of 'com.mapbox.navigation.ui.app.internal.SharedApp' appears in /data/app/~~A22N4mjD5b-l0vUK9le73A==/com.mapbox.navigation.qa_test_app-7lYqGWAs7nQNNQ-aIPqa3w==/base.apk!classes12.dex)
        at com.mapbox.androidauto.MapboxCarApp.setup(MapboxCarApp.kt:51)
        at com.mapbox.navigation.qa_test_app.QaTestApplication.onCreate(QaTestApplication.kt:11)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1223)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6762)
        at android.app.ActivityThread.access$1500(ActivityThread.java:256)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2091)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7870)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

Reply:
It is hard to maintain compatibility of `SharedApp`, as it is not tracked by Metalava.

Reply:
There is also this issue https://github.com/mapbox/mapbox-navigation-android/issues/6253

So I think we could make `SharedApp` backwards compatible and then we'll be good to go.

Reply:
> It is hard to maintain compatibility of SharedApp, as it is not tracked by Metalava.

I think we can do it until we have a public api. Here's a step https://github.com/mapbox/mapbox-navigation-android/pull/6303